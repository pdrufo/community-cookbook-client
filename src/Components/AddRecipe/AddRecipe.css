.add-recipe{
  margin-top: 100px;
  color: rgb(26, 29, 80);
  font-family: 'Libre Baskerville', serif;
  font-size: 20px;
}
form {
  text-align: left;
}

input, textarea {
  width: 75%;
  background-color: #2982C2;
}


label {
  display: block;
  font-size: 16px;
  margin-bottom: 10px;
}

.form-section {
  margin-bottom: 10px;
  text-align: center;
  color: rgb(26, 29, 80);
  font-family: 'Libre Baskerville', serif;
}
button {
  margin-bottom: 50px;
  padding: 10px;
  background-color: #1FC2AF;
  border-radius: 5px;
}
.reset-button {
  margin-left: 10px;
}
::placeholder {
  color:white;
  font-size: 12px;
}

@media only screen and (min-width: 768px){
  .add-recipe{
    margin-top: 125px;
    font-size: 30px;
  }
  
  input, textarea {
    width: 80%;
  }
  label {
    font-size: 20px;
    margin-bottom: 10px;
  }
  
  ::placeholder {
    font-size: 14px;
    vertical-align: top;
  }
  
  button {
    padding: 30px;
    background-color: #1FC2AF;
    border-radius: 5px;
    font-size: 20px;
  }
  .reset-button {
    margin-left: 20px;
  }
  .form-section {
    margin-bottom: 40px;
  }
}

@media only screen and (min-width: 1200px){
  
  
  input, textarea {
    width: 40%;
  }
 

  
  button {
    padding: 20px;
  }
  
}

