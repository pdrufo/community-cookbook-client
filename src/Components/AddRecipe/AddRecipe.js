import React from "react";
import "./AddRecipe.css";
import ApiContext from "../../ApiContext";
import config from "../../config";

/**AddRecipe will allow a user to input values into a form and POST the new recipe to the recipe database */
export default class AddRecipe extends React.Component {
  static defaultProps = {
    history: {
      push: () => {},
    },
  };

  static contextType = ApiContext;

  handleSubmit = (e) => {
    e.preventDefault();
    const newRecipe = {
      title: e.target["recipe-title"].value,
      ingredients: e.target["recipe-ingredients"].value,
      instructions: e.target["recipe-instructions"].value,
      source: e.target["recipe-source"].value,
    };

    if (
      !newRecipe.title ||
      !newRecipe.ingredients ||
      !newRecipe.instructions ||
      !newRecipe.source
    ) {
      alert(
        "Recipe title, ingredients, instructions, and source are all required"
      );
    } else {
      fetch(`${config.API_ENDPOINT}/recipes`, {
        method: "POST",
        headers: {
          "content-type": "application/json",
        },
        body: JSON.stringify(newRecipe),
      })
        .then((res) => {
          if (!res.ok) return res.json().then((e) => Promise.reject(e));
          return res.json();
        })
        .then((recipe) => {
          this.context.addRecipe(recipe);
          this.props.history.push(`/recipes`);
        })
        .catch((error) => {
          console.error({ error });
        });
    }
  };
  render() {
    return (
      <section className="add-recipe">
        <h3> Add New Recipe</h3>
        <form className="add-recipe-form" onSubmit={this.handleSubmit}>
          <div className="form-section">
            <label htmlFor="recipe-title">
              Recipe Title
              <input
                aria-label="input for label"
                type="text"
                name="recipe-title"
                placeholder="Chicken Picatta"
                required
              />
            </label>
          </div>

          <div className="form-section">
            <label htmlFor="recipe-ingredients">
              Add Ingredients
              <textarea
                aria-label="input for label"
                name="recipe-ingredients"
                rows="5"
                placeholder="separate ingredients by comma"
                required
              />
            </label>
          </div>

          <div className="form-section">
            <label htmlFor="recipe-instructions">
              Add instructions
              <textarea
                aria-label="input for label"
                name="recipe-instructions"
                rows="5"
              />
            </label>
          </div>

          <div className="form-section">
            <label htmlFor="recipe-source">
              Add Source
              <textarea
                aria-label="input for label"
                name="recipe-source"
                rows="5"
                placeholder="e.g. NYTimes Food"
                required
              />
            </label>
          </div>

          <div className="form-section">
            <button type="submit">Submit</button>
            <button type="reset" className="reset-button">
              Reset
            </button>
          </div>
        </form>
      </section>
    );
  }
}
