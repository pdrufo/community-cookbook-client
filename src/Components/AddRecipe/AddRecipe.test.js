import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom'
import AddRecipe from '././AddRecipe';
import renderer from 'react-test-renderer'


describe('Add Recipe' , () => {


  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter><AddRecipe /></BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<BrowserRouter><AddRecipe /></BrowserRouter>)
      .toJSON()
    expect(tree).toMatchSnapshot();
  })

  
})