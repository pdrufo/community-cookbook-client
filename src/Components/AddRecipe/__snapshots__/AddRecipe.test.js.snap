// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Recipe renders the UI as expected 1`] = `
<section
  className="add-recipe"
>
  <h3>
     Add New Recipe
  </h3>
  <form
    className="add-recipe-form"
    onSubmit={[Function]}
  >
    <div
      className="form-section"
    >
      <label
        htmlFor="recipe-title"
      >
        Recipe Title
      </label>
      <input
        name="recipe-title"
        placeholder="Chicken Picatta"
        required={true}
        type="text"
      />
    </div>
    <div
      className="form-section"
    >
      <label
        htmlFor="recipe-ingredients"
      >
        Add Ingredients
      </label>
      <textarea
        name="recipe-ingredients"
        placeholder="separate ingredients by comma"
        required={true}
        rows="5"
      />
    </div>
    <div
      className="form-section"
    >
      <label
        htmlFor="recipe-instructions"
      >
        Add Instructions
      </label>
      <textarea
        name="recipe-instructions"
        required={true}
        rows="5"
      />
    </div>
    <div
      className="form-section"
    >
      <label
        htmlFor="recipe-source"
      >
        Add Source
      </label>
      <textarea
        name="recipe-source"
        placeholder="e.g. NYTimes Food"
        required={true}
        rows="5"
      />
    </div>
    <div
      className="form-section"
    >
      <button
        type="submit"
      >
        Submit
      </button>
      <button
        type="reset"
      >
        Reset
      </button>
    </div>
  </form>
</section>
`;
